openstf_dbc/data_interface.py

```diff
@@ -8,29 +8,46 @@
 import requests
 import sqlalchemy
 
-from openstf_dbc.config.config import ConfigManager
+from openstf_dbc import Singleton
 from openstf_dbc.ktp_api import KtpApi
 from openstf_dbc.log import logging
 
 # Define abstract interface
 
 
-class _DataInterface:
+class _DataInterface(metaclass=Singleton):
+    def __init__(self, config):
+        """Generic data interface.
 
-    _instance = None
+        All connections and queries to the influx database, mysql databases and
```

Comment:
Nice to also document the input argument here and a bit about what is expected

Reply:
Done, added documentation to both the database and datainterface constructors.